diod/ops.c

```diff
@@ -74,7 +74,9 @@
 #include <pthread.h>
 #include <errno.h>
 #include <sys/types.h>
+#ifndef __FreeBSD__
 #include <sys/sysmacros.h>
+#endif
```

Comment:
It would be better to handle this by adding `sys/symacros.h` to AC_CHECK_HEADERS in `configure.ac`, then using `#if HAVE_SYS_SYSMACROS_H`.

(I'd like to avoid direct platform ifdefs in the code if possilble)

Reply:
I'm not very used to autoconf, but here it is.  Thanks for explaining how to do it!